fixtures/interactive-dev-tests/tests/index.test.ts

```diff
@@ -290,29 +290,45 @@ it.each(exitKeys)("multiworker cleanly exits with $name", async ({ key }) => {
 	}
 });
 
-// it seems like if we spam the container too often, it freezes up and crashes
+// it seems like if we hit to container when the port is ready but the app inside is not listening, the fetch will hang
+async function fetchWithTimeout(input: RequestInfo) {
+	const controller = new AbortController();
+	const { signal } = controller;
+	setTimeout(3_000).then(() => {
+		controller.abort();
+	});
+	return await fetch(input, {
+		signal,
+		headers: { "MF-Disable-Pretty-Error": "true" },
+	});
+}
+
 const WAITFOR_OPTIONS = { timeout: 2000, interval: 500 };
 baseDescribe.skipIf(process.platform !== "linux" && process.env.CI === "true")(
 	"container dev",
-	{ retry: 0, timeout: 90000 },
+	{ retry: 3, timeout: 90000 },
 	() => {
 		let tmpDir: string;
+
 		beforeAll(async () => {
 			tmpDir = fs.mkdtempSync(path.join(tmpdir(), "wrangler-container-"));
+			const ids = getContainerIds();
+			if (ids.length > 0) {
+				execSync("docker rm -f " + ids.join(" "), {
+					encoding: "utf8",
+				});
+			}
+		});
+
+		// reset any rewrites to the container app
+		afterEach(() => {
 			fs.cpSync(
 				path.resolve(__dirname, "../", "container-app"),
 				path.join(tmpDir),
 				{
 					recursive: true,
 				}
 			);
-
-			const ids = getContainerIds();
-			if (ids.length > 0) {
-				execSync("docker rm -f " + ids.join(" "), {
-					encoding: "utf8",
-				});
```

Comment:
Doesn't moving this mean that we will leave containers behind after the test run?